File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n=nums.size();
        for(int num:nums) {
            int idx=abs(num)-1;
            if(nums[idx]>0) 
                nums[idx]*=-1;
        }
        vector<int> result;
        for (int i=0;i<n;i++) {
            if(nums[i]>0) 
                result.push_back(i+1);
        }
        return result;
    }
};

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums):
        n = len(nums)

        for i in range(n):
            index = abs(nums[i]) - 1
            if nums[index] > 0:
                nums[index] *= -1

        result = []
        for i in range(n):
            if nums[i] > 0:
                result.append(i + 1)

        return result
